backends/tofino/bf-p4c/mau/instruction_selection.cpp

```diff
@@ -408,7 +408,7 @@ const IR::Node *Synth2PortSetup::postorder(IR::MAU::Primitive *prim) {
     auto dot = prim->name.find('.');
     auto objType = dot ? prim->name.before(dot) : cstring();
     const char *tail = objType.c_str() + objType.size();
-    while (tail > objType && std::isdigit(tail[-1])) --tail;
+    while (tail > objType.c_str() && std::isdigit(tail[-1])) --tail;
```

Comment:
This is comparing char pointers now, no? Maybe the intention was to compare the strings.

Reply:
Maybe, but previously it worked because of implicit conversion of `cstring` to `const char*`. And I think here it is comparing pointers: it just iterates over `cstring` backwards skipping digits.